console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/client/device/history/DeviceTabHistory.java

```diff
@@ -103,11 +102,7 @@ public DeviceTabHistory(GwtSession currentSession) {
     @Override
     public void setEntity(GwtDevice gwtDevice) {
         super.setEntity(gwtDevice);
-        setEnabled(gwtDevice != null &&
-                gwtDevice.isOnline() &&
-                currentSession.hasPermission(DeviceManagementSessionPermission.read()) &&
-                (gwtDevice.hasApplication(GwtDevice.GwtDeviceApplication.APP_DEPLOY_V1) || (gwtDevice.hasApplication(GwtDevice.GwtDeviceApplication.APP_DEPLOY_V2))));
-
+        setEnabled(gwtDevice != null);
```

Comment:
IMHO I don't like this kind of coding, especially as one line before gwtDevice is set.
It would be much nicer if both lines would be enclosed in if and setEnabled(true).